napari/layers/shapes/shapes.py

```diff
@@ -1369,9 +1387,20 @@ def mode(self, mode):
         ]
 
         self.events.mode(mode=mode)
-        if not (mode in draw_modes and old_mode in draw_modes):
-            self._finish_drawing()
-        self.refresh()
+        # only update the thumbnail if there are < 100 shapes
+        if len(self.data) > self._thumbnail_update_thresh:
```

Comment:
See comment above, I think it's cleaner if we just always do the block thumbnail path here AND additionally have a catch if we have > 100 shapes to pick those 100 randomly like we do in vectors

Reply:
Changing mode should also never need a thumbnail update, as changing mode shouldn't change the data, so it's probably always worth blocking thumbnail generation here!!!

Reply:
> Thinking about this one more, changing mode should also never need a thumbnail update, as changing mode shouldn't change the data, so it's probably always worth blocking thumbnail generation here!

That make sense to me. I will set it so that we always block thumbnail regeneration during mode changes.

Reply:
Done!